core/Test/Tasty/Run.hs

```diff
@@ -219,14 +221,21 @@ type Tr = Traversal
 
 -- | Exceptions related to dependencies between tests.
 data DependencyException
-  = DependencyLoop
-    -- ^ Test dependencies form a loop. In other words, test A cannot start
+  = DependencyLoop [[Path]]
+    -- ^ Test dependencies form cycles. In other words, test A cannot start
     -- until test B finishes, and test B cannot start until test
-    -- A finishes.
+    -- A finishes. Field lists detected cycles.
   deriving (Typeable)
 
 instance Show DependencyException where
-  show DependencyLoop = "Test dependencies form a loop."
+  show (DependencyLoop css) = "Test dependencies have cycles:" ++ showCycles css
```

Comment:
```suggestion
  show (DependencyLoop css) = "Test dependencies have cycles:\n" ++ showCycles css
```

Otherwise the output looks like 
```
mod-bench: Test dependencies have cycles:All.Power.Data.Mod, All.Power.Data.Mod
```

Reply:
Yes, you're right. And `showCycle` should do the `- ` stuff. Currently it looks like:

```
mod-bench: Test dependencies have cycles:All.Power.Data.Mod, All.Power.Data.Mod
- Foo, Bar, Foo
```